src/agni/client/view/ChatView.java

```diff
@@ -1,36 +1,166 @@
+import charvax.swing.JMenuItem;
+import charvax.swing.JTextArea;
+import charvax.swing.JTextField;
+
+public class ChatView extends JFrame implements AgniClientView, ActionListener, KeyListener {
+
+    private AgniClient client;
     private InfoRequestActionHandler infoRequestActionHandler;
     private UserActionHandler userActionHandler;
     private ChatActionHandler chatActionHandler;
     private FileActionHandler fileActionHandler;
-    private HeartbeatActionHandler heartbeatActionHandler;
-
-    public ChatView(InfoRequestActionHandler infoRequestActionHandler,
+    private HeartbeatMonitor serverMonitor;
+    private JTextField inputLine;
+    private JTextArea outputArea;
+    private final int WIDTH = 80;
+    private final int HEIGHT = 24;
+
+    public ChatView(AgniClient client,
+                    InfoRequestActionHandler infoRequestActionHandler,
                     UserActionHandler userActionHandler,
                     ChatActionHandler chatActionHandler,
-                    FileActionHandler fileActionHandler,
-                    HeartbeatActionHandler heartbeatActionHandler) {
+                    FileActionHandler fileActionHandler) {
 
+        this.client = client;
         this.infoRequestActionHandler = infoRequestActionHandler;
         this.userActionHandler = userActionHandler;
         this.chatActionHandler = chatActionHandler;
         this.fileActionHandler = fileActionHandler;
-        this.heartbeatActionHandler = heartbeatActionHandler;
+        this.serverMonitor = new HeartbeatMonitor(5000000000L); //5 second timeout period
+        inputLine = new JTextField("", WIDTH - 2);
+        outputArea = new JTextArea("", HEIGHT - 5, WIDTH - 2);
+        setupUi();
+    }
+
+    private void setupUi() {
+        Container contentPane = getContentPane();
+        contentPane.setLayout(new BorderLayout());
+
+        JMenuBar menubar = new JMenuBar();
+        JMenu jMenuFile = new JMenu("Agni");
+        JMenuItem jMenuItemFileExit = new JMenuItem("Exit");
+        jMenuItemFileExit.addActionListener((ActionListener) this);
+        jMenuFile.add(jMenuItemFileExit);
+        menubar.add(jMenuFile);
+        setJMenuBar(menubar);
+
+        contentPane.add(outputArea, BorderLayout.NORTH);
+        contentPane.add(new JLabel(">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"));
+        contentPane.add(inputLine, BorderLayout.SOUTH);
+
+        setLocation(0, 0);
+        setSize(WIDTH, HEIGHT);
+        validate();
+
+        outputArea.setEditable(false);
+        outputArea.setText("You have joined the chat!\n");
+        inputLine.addKeyListener(this);
+        // (new Thread(serverMonitor)).start();
     }
 
     @Override
-    public void heartbeatReaction() {
+    public void displayUi() {
+        show();
+        inputLine.requestFocus();
+        // appendToOutputArea("ChatView!");
+    }
+
+    public synchronized void appendToOutputArea(String message) {
```

Comment:
In the constructor. But append to output area gets called all over the place. Just hasn't been implemented in the reactions for `ChatView` yet.